Xamarin.Forms.Core/NavigationPage.cs

```diff
@@ -292,10 +299,13 @@ async Task<Page> INavigationPageController.RemoveAsyncInner(Page page, bool anim
 			EventHandler<NavigationRequestedEventArgs> requestPop = PopRequested;
 			if (requestPop != null)
 			{
+				_ignorePopCall = true;
```

Comment:
I agree this doesn't quite feel like the place to fix.  *RemoveAsyncInner* being called twice on the same page seems like the bug.  I wonder if we can fix this purely inside the iOS *NavigationRenderer* .  

For example returning from 
*NavigationRenderer.UpdateFormsInnerNavigation* if _ignorePopCall true

and then setting _ignorePopCall to true inside

*protected virtual async Task<bool> OnPopViewAsync(Page page, bool animated)*

 I also pushed a UI Test that we can use to test this as well

Reply:
agreed with @PureWeen

Reply:
@PureWeen your suggestion seems to work, I pushed the changes and tested with the UITest. But I don't think its an elegant solution 🤔

Reply:
@DerGary  I agree it's not the most elegant. There's a ping pong that happens unfortunately which I would say is from *RemoveAsyncInner* doing too much.  If the calls of remove start from iOS, via back button or swipe, they should probably call into something more specific on the shared code side. That being said *RemoveAsyncInner* is part of a public interface so we've kind of made our bed, so to speak, and doing it this way gives us our end though the means are slightly inelegant.

I've kicked off the UI Tests because that's where I mainly feel like we'll catch other side effects this might lead to